Library Catalog (Encore).js

```diff
@@ -36,204 +35,183 @@
-						availableItems[links[i].href] = links[i].textContent.trim();}
-					}
-			};
-			Zotero.selectItems(availableItems, function (items) {
-				if(!items) {
-					return true;
-				}
-				
-				var newUrls = new Array();
-				for(var itemURL in items) {
-					newUrls.push(itemURL.replace("?", "?marcData=Y&"));
-				}
-				pageByPage(marc, newUrls);
-			});
-		}
-	});
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function(items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			scrape(articles)
+		});
+	} else {
+		var marcURL = createMarcURL(url)
+		scrape([marcURL]);
+	}
 }
 
+function createMarcURL(url) {
+	//construct the marc URL
+	return url.replace(/\?/, "?marcData=Y&");
+}
 
+function scrape(marcURL) {
+	for (let i = 0; i < marcURL.length; i++) {
+		//Z.debug(marcURL[i])
+		//the library catalogue name isn't perfect, but should be unambiguous. 
+		var domain = marcURL[i].match(/https?:\/\/([^/]+)/);
+		ZU.doGet(marcURL[i], function(text) {
+			var translator = Zotero.loadTranslator("import");
+			translator.setTranslator("a6ee60df-1ddc-4aae-bb25-45e0537be973");
+			translator.getTranslatorObject(function(marc) {
+				var record = new marc.record();
+				var newItem = new Zotero.Item();
+
+				var linee = text.split("\n");
+				for (var i = 0; i < linee.length; i++) {
+					if (!linee[i]) {
+						continue;
+					}
 
-//functions:
-function scrape(marc, newDoc) {
-
-	var xpath = '//pre/text()';
-	if (newDoc.evaluate(xpath, newDoc, null, XPathResult.ANY_TYPE, null).iterateNext()) {
-		var elmts = newDoc.evaluate(xpath, newDoc, null, XPathResult.ANY_TYPE, null);
-		var useNodeValue = true;
-	} else {
-		var elmts = newDoc.evaluate('//pre', newDoc, null, XPathResult.ANY_TYPE, null);
-		var useNodeValue = false;
-	}
-
-	var elmt;
-	while(elmt = elmts.iterateNext()) {
-		if (useNodeValue) {
-			var text = elmt.nodeValue;
-		} else {
-			var text = elmt.textContent;
-		}
-		var newItem = new Zotero.Item();
-		var record = new marc.record();
-		
-		var linee = text.split("\n");
-		for (var i=0; i<linee.length; i++) {
-			if(!linee[i]) {
-				continue;
-			}
-			
-			linee[i] = linee[i].replace(/[\xA0_\t]/g, " ");
-			var value = linee[i].substr(7);
-			
-			if(linee[i].substr(0, 6) == "      ") {
-				// add this onto previous value
-				tagValue += value;
-			} else {
-				if(linee[i].substr(0, 6) == "LEADER") {
-					// trap leader
-					record.leader = value;
-				} else {
-					if(tagValue) {	// finish last tag
-						tagValue = tagValue.replace(/\|(.)/g, marc.subfieldDelimiter+"$1");
-						if(tagValue[0] != marc.subfieldDelimiter) {
-							tagValue = marc.subfieldDelimiter+"a"+tagValue;
+					linee[i] = linee[i].replace(/[\xA0_\t]/g, " ");
+					var value = linee[i].substr(7);
+
+					if (linee[i].substr(0, 6) == "      ") {
+						// add this onto previous value
+						tagValue += value;
```

Comment:
This is IMO quite confusing and `tagValue` is not yet initialized.

I suggest to actually clean the text before splitting into lines with something like:
```js
				// delete empty lines, delete confusing newlines, e.g. 
				// 245 10 Testing a theoretical model of critical thinking and 
                		//        cognitive development /|cby Jane Rapps 
				text = text.replace(/^\n/mg, '').replace(/\s?\n\s+/gm, ' ');
```